R/gs_spending_combo.R

```diff
@@ -30,9 +30,77 @@
+#' par <- list(sf = gsDesign::sfExponential, total_spend = 0.025, param = 1)
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
+#' # Two-parameter Spending Function Families
+#' par <- list(sf = gsDesign::sfLogistic, total_spend = 0.025, param = c(.1, .4, .01, .1))
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
+#' par <- list(sf = gsDesign::sfBetaDist, total_spend = 0.025, param = c(.1, .4, .01, .1))
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
+#' par <- list(sf = gsDesign::sfCauchy, total_spend = 0.025, param = c(.1, .4, .01, .1))
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
+#' par <- list(sf = gsDesign::sfExtremeValue, total_spend = 0.025, param = c(.1, .4, .01, .1))
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
+#' par <- list(sf = gsDesign::sfExtremeValue2, total_spend = 0.025, param = c(.1, .4, .01, .1))
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
+#' par <- list(sf = gsDesign::sfNormal, total_spend = 0.025, param = c(.1, .4, .01, .1))
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
+#' # t-distribution Spending Function
+#' par <- list(sf = gsDesign::sfTDist, total_spend = 0.025, param = c(-1, 1.5, 4))
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
+#' # Piecewise Linear and Step Function Spending Functions
+#' par <- list(sf = gsDesign::sfLinear, total_spend = 0.025, param = c(.2, .4, .05, .2))
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
+#' par <- list(sf = gsDesign::sfStep, total_spend = 0.025, param = c(1 / 3, 2 / 3, .1, .1))
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
+#' # Pointwise Spending Function
+#' par <- list(sf = gsDesign::sfPoints, total_spend = 0.025, param = c(.25, .25))
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
+#' # Truncated, trimmed and gapped spending functions
+#' par <- list(sf = gsDesign::sfTruncated, total_spend = 0.025, param = list(trange = c(.2, .8), sf = gsDesign::sfHSD, param = 1))
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
+#' par <- list(sf = gsDesign::sfTrimmed, total_spend = 0.025, param = list(trange = c(.2, .8), sf = gsDesign::sfHSD, param = 1))
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
+#' par <- list(sf = gsDesign::sfGapped, total_spend = 0.025, param = list(trange = c(.2, .8), sf = gsDesign::sfHSD, param = 1))
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
+#' # Xi and Gallo conditional error spending functions
+#' par <- list(sf = gsDesign::sfXG1, total_spend = 0.025, param = 0.5)
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
+#' par <- list(sf = gsDesign::sfXG2, total_spend = 0.025, param = 0.14)
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
+#' par <- list(sf = gsDesign::sfXG3, total_spend = 0.025, param = 0.013)
+#' gs_spending_combo(par, info = 1:3 / 3)
+#'
 #' # beta-spending
 #' par <- list(sf = gsDesign::sfLDOF, total_spend = 0.2)
 #' gs_spending_combo(par, info = 1:3 / 3)
 gs_spending_combo <- function(par = NULL, info = NULL, ...) {
-  par$sf(par$total_spend, info, ...)$spend
+  par$sf(alpha = par$total_spend, t = info, param = par$param, ...)$spend
```

Comment:
From [gsDesign technical manual](https://keaven.github.io/gsd-tech-manual/spendfn.html#sfArgs):

> ...a spending function passed to `gsDesign()` through the arguments (upper bound) and (lower bound) must have the following calling sequence: `sfname(alpha, t, param)`

and it looks like so for all existing spending functions in gsDesign. So maybe `...` is now redundant.

Reply:
Thanks for connecting to gsDesign, @nanxstats. I deleted the `....` and it works.